### ISSUE TITLE
[RA2 Core]: Build a consensus on bare metal containerization

### ISSUE THREAD

**Issue by CsatariGergely** [cntt-n/CNTT#453]
Sub chapter [A note on bare metal containerisation](https://github.com/cntt-n/CNTT/blob/master/doc/ref_arch/kubernetes/chapters/chapter01.md#a-note-on-bare-metal-containerisation) were merged before a full consensus would have been reached in #383 . 
The resolution of this issue should build a consensus on the bare metal containerization and document the result in a Chapter 02 of the Reference Architecture.

**Comment by tomkivlin** [comment]
From 31.10 meeting: @tomkivlin assigned to produce content.

**Comment by tomkivlin** [comment]
I think the approach I will take to this is to suggest additions to the Requirements that Pankaj has already started and it would make sense to add these once the existing PR is merged as to avoid any conflicts.  

I think it consists of, broadly speaking, the following:

2.2 - no change
2.3 - suggest that the opening statement includes something that makes it clear that a Kubernetes cluster can be deployed on virtual or physical infrastructure (aka bare metal) and that certain components within a cluster will deal with the management of virtual or physical compute, storage and network resources (e.g. container runtime, via the OS kernel, will manage virtual and physical compute resources for containerised software to consume).
2.3.1 - no change
2.3.2 - suggest the following addition, or something along these lines:

|Ref#|sub-category|Description|
|---|---|---|
|req.inf.vir.01|Virtualisation|The Architecture **must** support the capability for Containers to consume virtualised compute, storage and network resources.|
|req.inf.phy.01|Physical Infrastructure|The Architecture **must** support the capability for Containers to consume physical (aka bare metal) compute, storage and network resources.|

2.3.3 - I'm not sure anything needs changing or adding here
2.3.4 - no update needed, I think
2.3.5 - I think the following needs changing to a **must**

|Ref#|sub-category|Description|
|---|---|---|
|req.lcm.adp.04|Automated deployment|The Architecture should support declarative specifications of hardware and software assets for automated deployment, configuration, maintenance and management.|

2.3.6 - no update needed
2.3.7 - suggest the following is expanded (to cover kernel based isolation - i.e. no privileged containers) - I know it's being discussed in the Requirements meeting.

|Ref#|sub-category|Description|
|---|---|---|
|req.sec.gen.01|General|The Architecture must provide workload isolation.|

If this is agreed and applied, then a separate Issue can be created to address any bare metal requirements in the architecture chapters 3 and 4.  e.g. anything different or that has a potential impact (positive or negative) to virtualised containerisation - things like potential for increased performance, reduction in attack surface, impacts on multi-tenancy, workload isolation, etc.

**Comment by hdamker** [comment]
@tomkivlin Wasn't this issue raised to fixed the opionated section "### A note on virtualised and containerised workloads" in chapter 1? I can't find a respective merge request. Therefore I don't think it's done.

Regarding the added new requirements req.inf.vir.01 and req.inf.phy.01: I don't agree that both are "must" at the same time. The RA should not prescribe which kind of infrastructure provider a compliant RI is using. And I see no mandatory requirement to support both, virtualized and physical, at the same time. An operator can happily run all CNFs with one of the variants, if the Kubernetes RI is fulfilling all other requirements (think about K8s on vSphere, or a future AWS AKS compliant with CNTT RA2). An RI may support both physical and virtual, but not mandatory.

**Comment by tomkivlin** [comment]
@hdamker sorry - I meant to move to PR Review in progress - have corrected!

> The RA should not prescribe which kind of infrastructure provider a compliant RI is using. 

Totally agree. The intent (as per #562) is for RA2 to be able to run on **any** cloud.  However, there will be specific requirements or configuration that need to be made on each particular cloud in order to meet the requirements of the RM and of our RA2.  Therefore there will be a section in one of the chapters that details these specifics (e.g. on OpenStack we need to use x, configured in such a way to be able to deliver req.a.b.c).  Each RI will then choose which cloud to use, and will use the information in the RA2 to guide them with the implementation specifics.

My belief is that once #562 #452 and #570 are merged this issue will be fixed.  Would you agree, pending some clarification on the wording of the requirements in #452?  On which, I will comment in that PR.

**Comment by tomkivlin** [comment]
Based on a conversation at Kubecon today, I want to clarify what has been done and what I plan to do with this issue.

I think that the decision to use or not bare metal is not one that this RA2 should answer.  Nor do I think we should detail any guidance about when to use it or not.  However, I do think that that RA2 should include the ability to run on bare metal within the scope, requirements and any subsequent chapters (e.g. security, operations, etc.).  So my approach has been:
- Include requirements in #452 / #607 
- Describe the standalone nature of RA2 in the Ref Arch [README.md](https://github.com/cntt-n/CNTT/tree/master/doc/ref_arch)

The only thing I think might be worth doing is adding some wording to chapter01.md which explains all of this; specifically that the RA2 will include requirements to ensure an RI based on RA2 can run on bare metal, but it will not include guidance or decision-making criteria to guide operators as to whether or not they use bare metal.  Unless I have an objection in the next couple of days I'll prepare a PR.

@hdamker @fzdarsky @CsatariGergely

**Comment by tomkivlin** [comment]
PRs #607 and #745 have clarified this.
